controllers/webspherelibertyapplication_controller.go

```diff
@@ -657,11 +657,13 @@ func (r *ReconcileWebSphereLiberty) Reconcile(ctx context.Context, request ctrl.
 		reqLogger.V(1).Info(fmt.Sprintf("%s is not supported", prometheusv1.SchemeGroupVersion.String()))
 	}
 
-	// Delete completed Semeru instances because all pods now point to the newest Semeru service
-	if areCompletedSemeruInstancesMarkedToBeDeleted {
-		if err := r.deleteCompletedSemeruInstances(instance); err != nil {
-			reqLogger.Error(err, "Failed to delete completed Semeru instance")
-			return r.ManageError(err, common.StatusConditionTypeReconciled, instance)
+	if r.isWebSphereLibertyApplicationReady(instance) {
```

Comment:
It'll be good to do this check after checking areCompletedSemeruInstancesMarkedToBeDeleted, so status is not checked unnecessarily.

Reply:
Done